Combination of these glibc commits

b507b6d4c6 * Hurd: restore some SIOC ioctls
7073112263 * Hurd: comment ioctls which cannot currently compile
ec98b25af5 * Hurd: comment PF_ROUTE/AF_ROUTE defines
59a0de89ba * Hurd: comment PF_LINK/AF_LINK defines

From 59a0de89ba28028f5ec3f32c1da648a0d6f1dfc5 Mon Sep 17 00:00:00 2001
From: Pino Toscano <toscano.pino@tiscali.it>
Date: Tue, 26 May 2026 22:27:02 +0200
Subject: [PATCH 1/4] Hurd: comment PF_LINK/AF_LINK defines

Comment out the PF_LINK and AF_LINK defines, since they are usually associated
with struct sockaddr_dl, which is not available on Hurd.
---
 sysdeps/mach/hurd/bits/socket.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sysdeps/mach/hurd/bits/socket.h b/sysdeps/mach/hurd/bits/socket.h
index 9b81d48816..4a5807e619 100644
--- a/sysdeps/mach/hurd/bits/socket.h
+++ b/sysdeps/mach/hurd/bits/socket.h
@@ -96,7 +96,7 @@ enum __socket_type
 #define	PF_HYLINK	15	/* NSC Hyperchannel protocol.  */
 #define	PF_APPLETALK	16	/* Don't use this.  */
 #define	PF_ROUTE	17	/* Internal Routing Protocol.  */
-#define	PF_LINK		18	/* Link layer interface.  */
+/* #define	PF_LINK		18	Link layer interface.  */
 #define	PF_XTP		19	/* eXpress Transfer Protocol (no AF).  */
 #define	PF_COIP		20	/* Connection-oriented IP, aka ST II.  */
 #define	PF_CNT		21	/* Computer Network Technology.  */
@@ -129,7 +129,7 @@ enum __socket_type
 #define	AF_HYLINK	PF_HYLINK
 #define	AF_APPLETALK	PF_APPLETALK
 #define	AF_ROUTE	PF_ROUTE
-#define	AF_LINK		PF_LINK
+/* #define	AF_LINK		PF_LINK */
 #ifdef __USE_MISC
 # define	pseudo_AF_XTP	PF_XTP
 #endif
-- 
2.54.0

From ec98b25af5ae4b6dff24388ce10d0e64a7671363 Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Tue, 26 May 2026 22:28:49 +0200
Subject: [PATCH 2/4] Hurd: comment PF_ROUTE/AF_ROUTE defines

Comment out the PF_ROUTE and AF_ROUTE defines, since they would be used for
PF_ROUTE setsockopts, which are not available on Hurd.
---
 sysdeps/mach/hurd/bits/socket.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sysdeps/mach/hurd/bits/socket.h b/sysdeps/mach/hurd/bits/socket.h
index 4a5807e619..fb73dea063 100644
--- a/sysdeps/mach/hurd/bits/socket.h
+++ b/sysdeps/mach/hurd/bits/socket.h
@@ -95,7 +95,7 @@ enum __socket_type
 #define	PF_LAT		14	/* DEC Local Area Transport protocol.  */
 #define	PF_HYLINK	15	/* NSC Hyperchannel protocol.  */
 #define	PF_APPLETALK	16	/* Don't use this.  */
-#define	PF_ROUTE	17	/* Internal Routing Protocol.  */
+/* #define	PF_ROUTE	17	Internal Routing Protocol.  */
 /* #define	PF_LINK		18	Link layer interface.  */
 #define	PF_XTP		19	/* eXpress Transfer Protocol (no AF).  */
 #define	PF_COIP		20	/* Connection-oriented IP, aka ST II.  */
@@ -128,7 +128,7 @@ enum __socket_type
 #define	AF_LAT		PF_LAT
 #define	AF_HYLINK	PF_HYLINK
 #define	AF_APPLETALK	PF_APPLETALK
-#define	AF_ROUTE	PF_ROUTE
+/* #define	AF_ROUTE	PF_ROUTE */
 /* #define	AF_LINK		PF_LINK */
 #ifdef __USE_MISC
 # define	pseudo_AF_XTP	PF_XTP
-- 
2.54.0

From 707311226345f87ce6a1536a7d7365ce34b236b3 Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Tue, 26 May 2026 22:35:55 +0200
Subject: [PATCH 3/4] Hurd: comment ioctls which cannot currently compile

We don't currently have struct ifreq_short, ifreq_int and ifaliasreq, so
don't let applications even try to compile these.
---
 sysdeps/mach/hurd/bits/ioctls.h | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/sysdeps/mach/hurd/bits/ioctls.h b/sysdeps/mach/hurd/bits/ioctls.h
index 151761ef66..4588f813fb 100644
--- a/sysdeps/mach/hurd/bits/ioctls.h
+++ b/sysdeps/mach/hurd/bits/ioctls.h
@@ -293,8 +293,8 @@ enum __ioctl_datum { IOC_8, IOC_16, IOC_32, IOC_64 };
 #define	SIOCSIFDSTADDR	_IOW('i', 14, struct ifreq)	/* set p-p address */
 #define	OSIOCGIFDSTADDR	_IOWR('i',15, struct ifreq)	/* get p-p address */
 #define	SIOCGIFDSTADDR	_IOWR('i',34, struct ifreq)	/* get p-p address */
-#define	SIOCSIFFLAGS	_IOW('i', 16, struct ifreq_short)/* set ifnet flags */
-#define	SIOCGIFFLAGS	_IOWR('i',17, struct ifreq_short)/* get ifnet flags */
+/* #define	SIOCSIFFLAGS	_IOW('i', 16, struct ifreq_short) set ifnet flags */
+/* #define	SIOCGIFFLAGS	_IOWR('i',17, struct ifreq_short) get ifnet flags */
 #define	OSIOCGIFBRDADDR	_IOWR('i',18, struct ifreq)	/* get broadcast addr */
 #define	SIOCGIFBRDADDR	_IOWR('i',35, struct ifreq)	/* get broadcast addr */
 #define	SIOCSIFBRDADDR	_IOW('i',19, struct ifreq)	/* set broadcast addr */
@@ -303,21 +303,21 @@ enum __ioctl_datum { IOC_8, IOC_16, IOC_32, IOC_64 };
 #define	OSIOCGIFNETMASK	_IOWR('i',21, struct ifreq)	/* get net addr mask */
 #define	SIOCGIFNETMASK	_IOWR('i',37, struct ifreq)	/* get net addr mask */
 #define	SIOCSIFNETMASK	_IOW('i',22, struct ifreq)	/* set net addr mask */
-#define	SIOCGIFMETRIC	_IOWR('i',23, struct ifreq_int)	/* get IF metric */
-#define	SIOCSIFMETRIC	_IOW('i',24, struct ifreq_int)	/* set IF metric */
+/* #define	SIOCGIFMETRIC	_IOWR('i',23, struct ifreq_int)	 get IF metric */
+/* #define	SIOCSIFMETRIC	_IOW('i',24, struct ifreq_int)	 set IF metric */
 #define	SIOCDIFADDR	_IOW('i',25, struct ifreq)	/* delete IF addr */
-#define	SIOCAIFADDR	_IOW('i',26, struct ifaliasreq)	/* add/chg IF alias */
+/* #define	SIOCAIFADDR	_IOW('i',26, struct ifaliasreq)	 add/chg IF alias */
 
 #define	SIOCSARP	_IOW('i', 30, struct arpreq)	/* set arp entry */
 #define	OSIOCGARP	_IOWR('i',31, struct arpreq)	/* get arp entry */
 #define	SIOCGARP	_IOWR('i',38, struct arpreq)	/* get arp entry */
 #define	SIOCDARP	_IOW('i', 32, struct arpreq)	/* delete arp entry */
 
-#define SIOCGIFMTU	_IOWR('i', 51, struct ifreq_int)/* get IF mtu */
-#define SIOCSIFMTU	_IOW('i', 52, struct ifreq_int)	/* set IF mtu */
+/* #define SIOCGIFMTU	_IOWR('i', 51, struct ifreq_int) get IF mtu */
+/* #define SIOCSIFMTU	_IOW('i', 52, struct ifreq_int)	 set IF mtu */
 
-#define SIOCGIFINDEX	_IOWR('i', 90, struct ifreq_int)/* get IF index */
-#define SIOCGIFNAME	_IOWR('i', 91, struct ifreq_int)/* set IF name */
+/* #define SIOCGIFINDEX	_IOWR('i', 90, struct ifreq_int) get IF index */
+/* #define SIOCGIFNAME	_IOWR('i', 91, struct ifreq_int) set IF name */
 
 
 /* Compatibility with 4.3 BSD terminal driver.
-- 
2.54.0

From b507b6d4c6138bfa3d6ff04ff880dc626c1c3605 Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Tue, 26 May 2026 23:07:12 +0200
Subject: [PATCH 4/4] Hurd: restore some SIOC ioctls

We do not define struct ifreq_short and ifreq_int, but we do
define _IOT_ifreq_short and _IOT_ifreq_int, and need these for
sysdeps/gnu/ifaddrs.c and sysdeps/mach/hurd/if_index.c
---
 sysdeps/mach/hurd/bits/ioctls.h | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/sysdeps/mach/hurd/bits/ioctls.h b/sysdeps/mach/hurd/bits/ioctls.h
index 4588f813fb..162a815082 100644
--- a/sysdeps/mach/hurd/bits/ioctls.h
+++ b/sysdeps/mach/hurd/bits/ioctls.h
@@ -293,8 +293,8 @@ enum __ioctl_datum { IOC_8, IOC_16, IOC_32, IOC_64 };
 #define	SIOCSIFDSTADDR	_IOW('i', 14, struct ifreq)	/* set p-p address */
 #define	OSIOCGIFDSTADDR	_IOWR('i',15, struct ifreq)	/* get p-p address */
 #define	SIOCGIFDSTADDR	_IOWR('i',34, struct ifreq)	/* get p-p address */
-/* #define	SIOCSIFFLAGS	_IOW('i', 16, struct ifreq_short) set ifnet flags */
-/* #define	SIOCGIFFLAGS	_IOWR('i',17, struct ifreq_short) get ifnet flags */
+#define	SIOCSIFFLAGS	_IOW('i', 16, struct ifreq_short)/* set ifnet flags */
+#define	SIOCGIFFLAGS	_IOWR('i',17, struct ifreq_short)/* get ifnet flags */
 #define	OSIOCGIFBRDADDR	_IOWR('i',18, struct ifreq)	/* get broadcast addr */
 #define	SIOCGIFBRDADDR	_IOWR('i',35, struct ifreq)	/* get broadcast addr */
 #define	SIOCSIFBRDADDR	_IOW('i',19, struct ifreq)	/* set broadcast addr */
@@ -303,8 +303,8 @@ enum __ioctl_datum { IOC_8, IOC_16, IOC_32, IOC_64 };
 #define	OSIOCGIFNETMASK	_IOWR('i',21, struct ifreq)	/* get net addr mask */
 #define	SIOCGIFNETMASK	_IOWR('i',37, struct ifreq)	/* get net addr mask */
 #define	SIOCSIFNETMASK	_IOW('i',22, struct ifreq)	/* set net addr mask */
-/* #define	SIOCGIFMETRIC	_IOWR('i',23, struct ifreq_int)	 get IF metric */
-/* #define	SIOCSIFMETRIC	_IOW('i',24, struct ifreq_int)	 set IF metric */
+#define	SIOCGIFMETRIC	_IOWR('i',23, struct ifreq_int)	/* get IF metric */
+#define	SIOCSIFMETRIC	_IOW('i',24, struct ifreq_int)	/* set IF metric */
 #define	SIOCDIFADDR	_IOW('i',25, struct ifreq)	/* delete IF addr */
 /* #define	SIOCAIFADDR	_IOW('i',26, struct ifaliasreq)	 add/chg IF alias */
 
@@ -313,11 +313,11 @@ enum __ioctl_datum { IOC_8, IOC_16, IOC_32, IOC_64 };
 #define	SIOCGARP	_IOWR('i',38, struct arpreq)	/* get arp entry */
 #define	SIOCDARP	_IOW('i', 32, struct arpreq)	/* delete arp entry */
 
-/* #define SIOCGIFMTU	_IOWR('i', 51, struct ifreq_int) get IF mtu */
-/* #define SIOCSIFMTU	_IOW('i', 52, struct ifreq_int)	 set IF mtu */
+#define SIOCGIFMTU	_IOWR('i', 51, struct ifreq_int)/* get IF mtu */
+#define SIOCSIFMTU	_IOW('i', 52, struct ifreq_int)	/* set IF mtu */
 
-/* #define SIOCGIFINDEX	_IOWR('i', 90, struct ifreq_int) get IF index */
-/* #define SIOCGIFNAME	_IOWR('i', 91, struct ifreq_int) set IF name */
+#define SIOCGIFINDEX	_IOWR('i', 90, struct ifreq_int)/* get IF index */
+#define SIOCGIFNAME	_IOWR('i', 91, struct ifreq_int)/* set IF name */
 
 
 /* Compatibility with 4.3 BSD terminal driver.
-- 
2.54.0

